app/services/complaint_case_updater_service.rb

```diff
@@ -0,0 +1,37 @@
+class ComplaintCaseUpdaterService < CaseUpdaterService
+  attr_reader :result
+
+  INITAL_CASE_STATUS = 'to_be_assessed'
+
+  def call
+    reset_status_on_complaint_type_change
+    super
```

Comment:
The idea of having different update services for different situations is very good,  for this particular one,  if the current state is reset back to the initial one ( even different one), IMO the change needs to be reflected in the case-transitions as well otherwise the state from most recent record in case-transitions won't be matched, we can discuss it further if need